pkg/types/stack.go

```diff
@@ -1,20 +1,41 @@
 package types
 
 import (
+	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/stacks/pkg/compose/types"
+	"github.com/docker/docker/api/types/swarm"
 )
 
-// Stack represents a runtime instantiation of a Docker Compose based application
-type Stack struct {
-	Version
-	Spec           StackSpec          `json:"spec"`
-	StackResources StackResources     `json:"stack_resources"`
-	Orchestrator   OrchestratorChoice `json:"orchestrator"`
-	Status         StackStatus        `json:"stack_status"`
+const (
+	// StackEventType is the value of Type in an events.Message for stacks
+	StackEventType = "stack"
+	// StackLabel is a label on objects indicating the stack that it belongs to
+	StackLabel = "com.docker.stacks.stack_id"
+)
 
-	// TODO - temporary (not in swagger)
+// Stack represents a Stack with all of its elements converted to Engine
+// API types.
+// NOTE: Stack is only used internally for reconciliation of Swarm
+// stacks and is never exposed via the API.
```

Comment:
We should rewrite this comment, because it reflects the old understanding of "stack"